lib/fastly_nsq.rb

```diff
@@ -10,15 +10,51 @@
 require 'fastly_nsq/version'
 
 module FastlyNsq
-  def self.logger=(logger)
+  module_function
+
+  def channel=(channel)
+    @channel ||= channel
+  end
+
+  def logger=(logger)
     strategy.logger = logger
   end
 
-  def self.logger
+  def self.preprocessor=(preprocessor)
+    @preprocessor ||= preprocessor
+  end
```

Comment:
Why is this here? Aren't preprocessors a pre listener concept? Or is this to setup a global pre-processor?